.github/workflows/build.yaml

```diff
@@ -51,3 +40,62 @@ jobs:
           cache: 'npm'
       - run: npm ci
       - run: npm run build
+      - uses: actions/upload-artifact@v2
+        with:
+          name: dist
+          path: dist
+  e2e:
+    runs-on: ubuntu-latest
+    needs:
+      - build
+      - lint
+      - test
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: "0"
+      - uses: actions/download-artifact@v2
+        with:
+          name: dist
+          path: dist
+      - uses: cypress-io/github-action@v2
+        with:
+          start: npm run http-server
+      - uses: actions/upload-artifact@v2
+        with:
+          name: e2e-videos
+          path: cypress/videos
+  docker:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    needs:
+      - e2e
```

Comment:
Why does the docker build need e2e test?
We could save time by just depending on `test` only

Reply:
I changed the `need` to `build`.